script/make_release.py

```diff
@@ -0,0 +1,463 @@
+
+    if not success:
+        os.remove(archive_path)
+        raise Exception('The CRC check for {} failed.'.format(archive_path))
+
+    LOGGER.info('Created archive {}.'.format(archive_path))
+
+def run_bench(stockfish):
+    command = [stockfish, 'bench']
+    if SDE_PATH:
+        command = [SDE_PATH, '--'] + command
+
+    proc = subprocess.run(command, stderr=subprocess.PIPE, stdout=subprocess.DEVNULL, universal_newlines=True)
+    return int([line for line in proc.stderr.splitlines() if 'Nodes searched' in line][0].split(':')[1].strip())
+
+def compile_target(
+    *,
+    repo_dir,
+    out_dir,
+    build_type='profile-build',
+    arch='x86-64-modern',
+    prefix='',
+    expected_bench=None,
+    num_threads=1,
+    compiler=None):
+    '''
+    This function performs the following:
+    1. Copies the contents of `repo_dir` into a temporary directory.
+    2. Runs `make clean` in that temporary directory + ./src/
+    3. Runs `make `build_type`` for the specified `arch`.
+       NOTE: If SDE_PATH is not empty then SDE_PATH is set for all make commands.
+    4. Copies the resulting executable file into `out_dir` with a given `prefix`
+    '''
+    assert num_threads >= 1
+
+    if prefix != '':
+        prefix = '-' + prefix
+
+    if args.keep_logs:
+        # If the user wants to keep logs a file is created and stdout redirected to it.
+        # Otherwise we use the dummy context manager and stdout is not redirected.
+        logfile_name = os.path.join(out_dir, 'stockfish{}-{}-{}.log'.format(prefix, build_type, arch))
+        LOGGER.info('{}: Creating log file at {}.'.format(arch, logfile_name))
+
+        logfile = open(logfile_name, 'w')
+        stdout = logfile
+    else:
+        logfile = DummyContextManager()
+        stdout = STDOUT
+
+    with logfile:
+        with TemporaryDirectory('./stockfish-{}-{}-'.format(build_type, arch)) as copy_dir:
+            src_dir = os.path.join(copy_dir.path, 'src')
+
+            LOGGER.info('{}: Copying original files from {} to {}'.format(arch, repo_dir, copy_dir.path))
+
+            shutil.copytree(os.path.join(repo_dir, 'src'), os.path.join(copy_dir.path, 'src'))
+
+            LOGGER.info('{}: Running `make clean`'.format(arch))
+
+            subprocess.run( \
+                ['make', 'clean'], \
+                cwd=src_dir, stdout=stdout, stderr=STDERR)
+
+            # We form the make command used to build stockfish.
+            # If SDE_PATH is given the only thing that needs to be done is
+            # it needs to be passed to the make script.
+            # There is no downside to always running under SDE if provided.
+            make_command = ['make', build_type, 'ARCH={}'.format(arch)]
+            if SDE_PATH:
+                make_command += ['SDE_PATH="{}"'.format(SDE_PATH)]
+            if compiler:
+                make_command += ['COMP={}'.format(compiler)]
+            if num_threads > 1:
+                make_command += ['-j{}'.format(num_threads)]
```

Comment:
@ppigazzini probably the problem is somewhere here? The "escaping" on the `SDE_PATH` might be incorrect, and it ends up containing `COMP=...` stuff too.

Reply:
Switching single quote - double quote at line 282 seems to solve the problem (???)
`make_command += ["SDE_PATH='{}'".format(SDE_PATH)]`

here is the ouput:
```
Step 2/4. Running benchmark for pgo-build ...
C:/Intel/sde-win/sde.exe -- ./stockfish.exe bench > /dev/null
```